src/main/java/org/extism/chicory/sdk/Manifest.java

```diff
@@ -34,32 +41,85 @@ class ManifestWasmUrl extends ManifestWasm {
     public ManifestWasmUrl(String url) {
         this.url = url;
     }
+
+    public InputStream getUrlAsStream() {
+        try {
+            var url = new URL(this.url);
+            return url.openStream();
+        } catch (MalformedURLException e) {
+            throw new ExtismException(e);
+        } catch (IOException e) {
+            throw new ExtismException(e);
+        }
+
+    }
 }
 
 public class Manifest {
-    final ManifestWasm[] wasms;
 
-    public static Manifest fromPath(String path) {
-        var wasm = new ManifestWasmPath(path);
-        return new Manifest(new ManifestWasm[]{wasm});
+    public enum Validation {
+        Import, Type, All;
     }
 
-    public static Manifest fromUrl(String url) {
-        var wasm = new ManifestWasmUrl(url);
-        return new Manifest(new ManifestWasm[]{wasm});
-    }
+    public static class Options {
+        boolean aot;
+        EnumSet<Validation> validationFlags = EnumSet.noneOf(Validation.class);
+
+        public Options withAoT() {
+            this.aot = true;
+            return this;
+        }
 
-    public static Manifest fromFilePath(Path path) {
-        var wasm = new ManifestWasmFile(path);
-        return new Manifest(new ManifestWasm[]{wasm});
+        public Options withValidation(Validation... vs) {
+            this.validationFlags.addAll(List.of(vs));
+            return this;
+        }
     }
 
-    public static Manifest fromBytes(byte[] bytes) {
-        var wasm = new ManifestWasmBytes(bytes);
-        return new Manifest(new ManifestWasm[]{wasm});
+    public static class Builder {
+        final ManifestWasm[] wasms;
+        private Options options;
+
+        private Builder(ManifestWasm[] manifestWasms) {
+            this.wasms = manifestWasms;
```

Comment:
One thing to note for a later PR, the Java SDK uses libextism. it serializes this manifest and let's libextism figure out how to link up all these modules. This SDK (along with the js-sdk and the go-sdk) don't have the liberty of re-using that code. So we'll need to borrow the implementation from one of those. This can be done in a later PR just thought i'd bring it up as i noticed it. Also to point out that, even though the interface will look like the Java SDK here, what we're doing is more akin to what the go and js sdk are doing (as non libextism sdks)